### PR TITLE
Use Spotifyd instead of vollibrespot to get latest librespot changes

### PR DESCRIPTION
**Pull request by klay2000**

### CODE REVIEW

File: amplipi/streams.py [hunk 1]
```diff
@@ -257,80 +259,68 @@ def connect(self, src):
...
       self._connect(src)
     except Exception as exc:
       print(f'error starting spotify: {exc}')
 
   def disconnect(self):
     if self._is_running():
-      os.killpg(os.getpgid(self.proc.pid), signal.SIGKILL)
-      self.proc2.kill()
+      self.proc.kill()
     self._disconnect()
```
Comment (linknum23) [inline amplipi/streams.py:287]:
```suggestion
    self._disconnect()
      self.connect_port = None
      self.mpris = None
```

File: amplipi/streams.py [hunk 2]
```diff
@@ -257,80 +259,68 @@ def connect(self, src):
...
-        if d['img_url']: # report generic spotify image in place of unspecified album art
-          source.img_url = d['img_url']
+      md = self.mpris.metadata()
+
+      source.state = 'playing' if self.mpris.is_playing() else 'paused'
+      source.artist = md.artist
+      source.track = md.title
+      source.album = md.album
+      source.supported_cmds=list(self.supported_cmds.keys())
+      if md.art_url: # report generic spotify image in place of unspecified album art
```
Comment (linknum23) [inline amplipi/streams.py:305]:
```suggestion
      if md.art_url:
```

File: amplipi/streams.py [hunk 3]
```diff
@@ -257,80 +259,68 @@ def connect(self, src):
...
     self.proc = None
-    self.proc2 = None
 
   def info(self) -> models.SourceInfo:
-    src_config_folder = f'{utils.get_folder("config")}/srcs/{self.src}'
-    loc = f'{src_config_folder}/currentSong'
     source = models.SourceInfo(
       name=self.full_name(),
       state=self.state,
       img_url='static/imgs/spotify.png'
```
Comment (linknum23) [inline amplipi/streams.py:294]:
```suggestion
      img_url='static/imgs/spotify.png' # report generic spotify image in place of unspecified album art
```

**Review by linknum23 — COMMENTED** [review]
Looks good! Lets test it.
I can help you add the installer for this in `scripts/configure.py` when we get there.

**Comment by linknum23** [comment]
Note most of those changes are only tangentally related to your PR just leftover things to clean up

File: requirements.txt [hunk 4]
```diff
@@ -1,8 +1,10 @@
 adafruit-circuitpython-rgb-display
 aiofiles
+dasbus
 deepdiff
 fastapi
 fastapi_utils
+pygobject
```
Comment (Lohrer) [inline requirements.txt:7]: Alphabetize please!

Also, add to base/apt _os_deps in configure.py:
```
# pygobject dependencies (Spotifyd)
'libgirepository1.0-dev', 'libcairo2-dev',
```

File: amplipi/streams.py [hunk 5]
```diff
@@ -257,80 +259,70 @@ def connect(self, src):
...
-    self.metaport = 5030 + 2*src
+    self.connect_port = 4070 + 2*src
     with open(toml_useful, 'r') as TOML:
       data = TOML.read()
-      data = data.replace('AmpliPi_TEMPLATE', f'{self.name}')
-      data = data.replace("device = 'ch'", f"device = 'ch{src}'")
-      data = data.replace('5030', f'{self.metaport}')
+      data = data.replace('device_name_in_spotify_connect', f'{self.name}')
+      data = data.replace("alsa_audio_device", f"ch{src}")
+      data = data.replace('1234', f'{self.connect_port}')
```
Comment (Lohrer) [inline amplipi/streams.py:267]: It would be awesome to expose `bitrate` and `autoplay` to the user if possible.

**Review by Lohrer — CHANGES_REQUESTED** [review]
Is there any way to download or build the `spotifyd` binary at install time, instead of adding to the repo?

File: amplipi/streams.py [hunk 6]
```diff
@@ -253,84 +255,83 @@ def connect(self, src):
     src_config_folder = f'{utils.get_folder("config")}/srcs/{src}'
     toml_template = f'{utils.get_folder("streams")}/spot_config.toml'
     toml_useful = f'{src_config_folder}/config.toml'
+
+    # make source folder
+    os.system(f'mkdir -p {src_config_folder}')
```
Comment (Lohrer) [inline amplipi/streams.py:260]: If this is a problem here, it's going to be a problem for all streams' connect() functions. Should this be done as part of the  `BaseStream` class, or in ctrl.py where it calls `stream.connect(idx)`, or somewhere else that handles this problem for all streams?

**Comment by linknum23** [comment]
Lets test this on Jason's house and see how we like it now that this has gotten past the multiple account crash bug.

File: scripts/configure.py [hunk 7]
```diff
@@ -85,6 +85,8 @@
              'xkcdpass',                    # Random passphrase generation
              'systemd-journal-remote',      # Remote/web based log access
              'jq',                          # JSON parser used in check-release script
+             # pygobject dependencies (Spotifyd)
+             'libgirepository1.0-dev', 'libcairo2-dev'
```
Comment (linknum23) [inline scripts/configure.py:89]: Looks like you will need to add these to `.github/workflows/python-app.yml` as well

File: amplipi/streams.py (hunk 5 above)
Comment (linknum23) [reply amplipi/streams.py:276]: Can we support higher bit rates?

**Comment by linknum23** [comment]
This has the potential to fix #247, #248, #214 and supersedes #318

File: amplipi/streams.py (hunk 6 above)
Comment (Lohrer) [reply amplipi/streams.py:265]: This was taken care of in #372

**Comment by linknum23** [comment]
This needs to be rebased on develop to get the hotfix for uvicorn

**Comment by linknum23** [comment]
Is this ready without volume adjustment?

**Comment by klay2000** [comment]
I need to change the cache directory, keep forgetting. I'll do it Monday then rebase and we can merge.

File: amplipi/streams.py (hunk 5 above)
Comment (linknum23) [reply amplipi/streams.py:270]: Let's do this in the future when we are able to better characterize utilization. We still need a 4th Spotify account!

**Pull request merged**